pepper-apis/src/main/java/org/broadinstitute/ddp/route/SendStudyEmailRoute.java

```diff
@@ -0,0 +1,85 @@
+import org.broadinstitute.ddp.db.dao.QueuedEventDao;
+import org.broadinstitute.ddp.db.dto.EventConfigurationDto;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.json.studyemail.Attachment;
+import org.broadinstitute.ddp.json.studyemail.SendStudyEmailPayload;
+import org.broadinstitute.ddp.model.activity.types.EventTriggerType;
+import org.broadinstitute.ddp.service.FileUploadService;
+import org.broadinstitute.ddp.util.ConfigManager;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import org.broadinstitute.ddp.util.ValidatedJsonInputRoute;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+public class SendStudyEmailRoute extends ValidatedJsonInputRoute<SendStudyEmailPayload> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SendStudyEmailRoute.class);
+
+    private final FileUploadService fileUploadService;
+
+
+    public SendStudyEmailRoute(FileUploadService fileUploadService) {
+        this.fileUploadService = fileUploadService;
+    }
+
+    @Override
+    public Object handle(Request request, Response response, SendStudyEmailPayload payload) throws Exception {
+        String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
+
+        TransactionWrapper.useTxn(handle -> {
+            Optional<Long> studyId = handle.attach(JdbiUmbrellaStudy.class).getIdByGuid(studyGuid);
+            if (studyId.isEmpty()) {
+                ApiError err = new ApiError(ErrorCodes.STUDY_NOT_FOUND, "Could not find study with guid " + studyGuid);
+                throw ResponseUtil.haltError(response, HttpStatus.SC_NOT_FOUND, err);
+            }
+            LOG.info("Handling study email sending in study {}", studyGuid);
+            List<EventConfigurationDto> eventConfigs =
+                    handle.attach(EventDao.class)
+                            .getNotificationConfigsForMailingListByEventType(studyGuid, EventTriggerType.SEND_STUDY_EMAIL);
+            if (!eventConfigs.isEmpty()) {
+                Set<Long> fileUploadIds = new HashSet<>();
+                List<Attachment> attachments = payload.getAttachments();
+                if (attachments != null) {
+                    for (Attachment attachment : attachments) {
+                        Long fileUploadId = fileUploadService.validateUploadAndGetId(handle, attachment.getGuid());
+                        if (fileUploadId == null) {
+                            String msg = String.format("Attachment %s is not verified", attachment.getGuid());
+                            throw ResponseUtil.haltError(response, HttpStatus.SC_BAD_REQUEST, msg);
+                        }
+                        fileUploadIds.add(fileUploadId);
+                    }
+                }
+                Config cfg = ConfigManager.getInstance().getConfig();
+                String toEmail = cfg.getString(ConfigFile.Sendgrid.TO_STUDY_EMAIL);
```

Comment:
What is the value of `TO_STUDY_EMAIL` in the configuration file? Remember, this config file is for the whole Pepper _platform_! We shouldn't put a study-specific email in here... I recommend querying the `sendgrid_configuration` table for the study's email to use as the "to email".

Reply:
Study-level configuration added